File: 001-TwoSum/TwoSum.cpp
#include <vector>
#include <unordered_map>
using namespace std;

class Solution {
    public:
    vector<int> twoSum(vector<int>& nums, int target){
        unordered_map<int, int> seen;
        for(int i = 0; i < nums.size(); i++){
            int complement = target - nums[i];
            if(seen.find(complement) != seen.end()){
                return {seen[complement], i};
            }else {
                seen[nums[i]] = i;
            }
        }
        return {};
    }
};

File: 001-TwoSum/TwoSum.java

import java.util.HashMap;

public class TwoSum {

    public class Solution {
    public int[] twoSum(int[] nums, int target){
        int[] indices = new int[2];
        HashMap<Integer, Integer> seen = new HashMap<>();
        for(int i = 0; i < nums.length; i++){
            Integer complement = target - nums[i];
            if(seen.containsKey(complement)){
                indices[0] = seen.get(complement);
                indices[1] = i;
                return indices;
            } else {
                seen.put(nums[i], i);
            }
        }
        return indices;
    }
}
}